### PR TITLE
✨roblox-games: added

### PR DESCRIPTION
**Pull request by eth0fox**
## Description

Add the Roblox Places plugin: Allows users to launch the Roblox platform from Raycast.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/74316107/153586557-feda6393-91ad-445a-8ffe-380db5acb641.mov

)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/roblox-games/package.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  "name": "roblox-games",
+  "title": "Roblox",
+  "description": "Searches for & launches Roblox places",
+  "icon": "command-icon.png",
+  "author": "TaylorRobinson",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Launch Roblox",
```
Comment (mattisssa) [inline extensions/roblox-games/package.json:12]: Judging by the screencast it seems like the command is more about games and searching for it. Should it be called something like
```suggestion
      "title": "Search Games",
```
?

File: extensions/roblox-games/package.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      "description": "Searches for & launches Roblox places",
```
Comment (mattisssa) [inline extensions/roblox-games/package.json:13]:
```suggestion
      "subtitle": "Roblox",
      "description": "Searches for & launches Roblox places",
```

File: extensions/roblox-games/package.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        },
+        {
+          "title": "Popular Worldwide",
+          "value": "Popular"
+        }
+      ]
+    },
+    {
+      "name": "accessory",
+      "title": "Accessory displays",
```
Comment (mattisssa) [inline extensions/roblox-games/package.json:51]:
```suggestion
      "title": "Accessory Displays",
```

File: extensions/roblox-games/src/auth.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { exec } from "child_process";
+import { promises as fs } from "fs";
+import fetch from "node-fetch";
+import { homedir } from "os";
+
+const configPath = homedir() + "/Library/Preferences/com.roblox.RobloxStudioBrowser.plist";
```
Comment (mattisssa) [inline extensions/roblox-games/src/auth.ts:6]: My main issue is that I don't have this file 😉

**Review by mattisssa — CHANGES_REQUESTED** [review]
Hey! I got an issue when I tried to run your extension.

1) Can we make a more proper text here in the error so users can understand what has happened?
2) Once the error happens, the loading indicator should be stopped 😉 
CleanShot 2022-02-14 at 14 05 38@2x

I also left a few suggestions below, please take a look at them

**Review by mattisssa — APPROVED** [review]

**Comment by mattisssa** [comment]
Hey @TayIorRobinson, seems like you have some prettier issues with your extension, could you please take a look at that?

Try to use this in `.eslintrc.json`:
```
{
  "root": true,
  "env": {
    "es2020": true,
    "node": true
  },
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"]
}
```

You have also a wrong Promise usage in the search file, at the place where you have `async`.

File: extensions/roblox-games/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+				"singleQuote": false,
+				"printWidth": 120,
+				"tabWidth": 4,
+				"useTabs": true
+			}
+		],
+		"no-unused-vars": "off",
+		"no-unused-expressions": "warn"
+	}
+}
```
Comment (mattisssa) [inline extensions/roblox-games/.eslintrc.json:35]:
```suggestion
{
  "root": true,
  "env": {
    "es2020": true,
    "node": true
  },
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"]
}
```

File: extensions/roblox-games/src/search.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import { requestIconForGames } from "./icons";
+
+let sortToken: Promise<string> | undefined;
+function getSortToken(cookie: string): Promise<string> {
+  if (sortToken)
+    return sortToken.catch(() => {
+      sortToken = undefined;
+      return getSortToken(cookie);
+    });
+  return (sortToken = new Promise(async (resolve, reject) => {
```
Comment (mattisssa) [inline extensions/roblox-games/src/search.ts:13]: I got the issue here for `async` usage. Could you take a look at this?

**Review by mattisssa — CHANGES_REQUESTED** [review]
Hey! I got a few issues during the review, please take a look at my comments below

**Comment by TayIorRobinson** [comment]
🚀 🎉 

Thanks for all your help @mattisssa!

**Pull request merged**